tests/test_00_unit/test_00_index_reader/__init__.py

```diff
@@ -0,0 +1,128 @@
+import unittest
+
+from ccp import index_reader
+
+
+class TestProject(unittest.TestCase):
+    """
+    Test the Project class representing an
+    object in container pipeline
+    """
+
+    def setUp(self):
+        self.entry = {
+            "id": "1",
+            "app-id": "foo",
+            "job-id": "bar",
+            "desired-tag": "latest",
+            "git-branch": "master",
+            "git-url": "https://github.com/someurl",
+            "git-path": "foo/",
+            "target-file": "Dockerfile",
+            "build-context": "./",
+            "notify-email": "abc@example/com",
+            "depends-on": ["centos/centos:latest", "centos/centos:7"],
+            "prebuild-script": "hooks/script.sh",
+            "prebuild-context": "/",
+        }
+        self.namespace = "ccp"
+
+    def test_load_entries(self):
+        """
+        Tests creating Project class object which loads
+        entries
+        """
+        index_reader.Project(self.entry, self.namespace)
+
+    def test_string_representation(self):
+        """
+        Test __str__ representation of Project object
+        """
+        project = index_reader.Project(self.entry, self.namespace)
+        self.assertEqual(
+            "foo-bar-latest",
+            project.__str__()
+        )
+
+    def test_replace_dot_slash_colon_(self):
+        """
+        Test the helper method to replace . / : _ with hyphen
+        """
+        project = index_reader.Project(self.entry, self.namespace)
+        self.assertEqual(
+            "a-b-c-d",
+            project.replace_dot_slash_colon_("a.b/c_d")
```

Comment:
This is great! But shouldn't we also check if `:` is also getting replaced as expected?

Reply:
Agreed, we either break it up into separate functions for each character or assert for each case.

Reply:
Ack, will change that. 

@mohammedzee1000 : sure! can be done.